apps/web/src/components/docs/Docs.tsx

```diff
@@ -25,21 +27,35 @@ const DOCS_WRAPPER_ELEMENT_ID = 'embedded-docs';
 type DocsProps = PropsWithChildren<
   DocsQueryResults & {
     isLoading?: boolean;
-    actions: ReactNode;
+    actions?: ReactNode;
+    isChildDocs?: boolean;
   }
 >;
 
 /*
  * Render the mdx for our mintlify docs inside of the web.
  * Fetching the compiled mdx from another service and then try to map the markdown to react components.
  */
-export const Docs = ({ code = '', description = '', title = '', isLoading, children, actions }: DocsProps) => {
+export const Docs = ({
+  code = '',
+  description = '',
+  title = '',
+  isLoading,
+  children: docsChildren,
+  actions = null,
+  isChildDocs = false,
+}: DocsProps) => {
   const Component = useMemo(() => {
     if (code.length === 0) {
       return null;
     }
 
     return getMDXComponent(code, {
+      framework: () => null,
+      NextStepsStep: () => <ChildDocs path="snippets/quickstart/next-steps" />,
+      TestStep: () => <ChildDocs path="snippets/quickstart/test" />,
+      DeployApp: () => <ChildDocs path="snippets/quickstart/deploy" />,
+      LocalStudio: () => <ChildDocs path="snippets/quickstart/start-studio" />,
```

Comment:
Will we need now for every snippet we are adding to modify the application? This might be happening frequently

Reply:
We will need to do this each time we import a new file that is not already mapped here.

Reply:
I agree will look into if we can auto resolve these on the backend